core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPathfinding.kt

```diff
@@ -207,9 +182,33 @@ private fun convertLocations(
     endBlocks: Collection<BlockId> = setOf()
 ): Set<EdgeLocation<STDCMEdge, STDCMEdge>> {
     val res = HashSet<EdgeLocation<STDCMEdge, STDCMEdge>>()
+    val fullInfra = graph.fullInfra
+
+    // Initialize the constraints
+    val loadingGaugeConstraints =
+        LoadingGaugeConstraints(fullInfra.blockInfra, fullInfra.rawInfra, listOf(rollingStock))
+    val electrificationConstraints =
+        ElectrificationConstraints(fullInfra.blockInfra, fullInfra.rawInfra, listOf(rollingStock))
+    val signalingSystemConstraints =
+        makeSignalingSystemConstraints(
+            fullInfra.blockInfra,
+            fullInfra.signalingSimulator,
+            listOf(rollingStock)
+        )
+    val combiner = ConstraintCombiner<BlockId, Block>()
```

Comment:
I'd prefer if the constraints were given as a parameter to this method, with the constraint instantiation being moved to a single method.

In fact said method could be shared with the block pathfinding, we (should) instantiate the same constraints in the same way.

Reply:
for context https://github.com/osrd-project/osrd/blob/dev/core/src/main/kotlin/fr/sncf/osrd/api/pathfinding/PathfindingBlocksEndpoint.kt#L215